public/app/features/dashboard/containers/DashboardPage.tsx

```diff
@@ -138,7 +138,16 @@ export class DashboardPage extends PureComponent<Props, State> {
     }
 
     if (prevProps.location.search !== this.props.location.search) {
-      getTimeSrv().updateTimeRangeFromUrl();
+      const prevUrlParams = new URLSearchParams(prevProps.location.search);
+      const urlParams = new URLSearchParams(this.props.location.search);
+
+      if (urlParams.get('from') !== prevUrlParams.get('from') && urlParams.get('to') !== prevUrlParams.get('to')) {
+        getTimeSrv().updateTimeRangeFromUrl();
+      }
+
+      if (!prevUrlParams.get('refresh') && urlParams.get('refresh')) {
+        getTimeSrv().setAutoRefresh(urlParams.get('refresh'));
+      }
```

Comment:
this feels unelegant, can we add from & to and refresh to DashboardPageRouteSearchParams? and thhen do 

if (preProps.queryParams.from !== this.props.queryParams.from ...)

And then I would update updateTimeRangeFromUrl to pass in from & to, so it does not need to ask it from locationService again)

Reply:
I have no idea what I've been thinking using the URLSearchParams here 🙈